ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/UserManagementServiceIT.scala

```diff
@@ -298,6 +299,129 @@ final class UserManagementServiceIT extends LedgerTestSuite {
+      res2 <- ledger.userManagement.listUsers(
+        ListUsersRequest(pageSize = 3, pageToken = res1.nextPageToken)
+      )
+      // Requesting last non-empty page of users
+      res3 <- ledger.userManagement.listUsers(
+        ListUsersRequest(pageSize = 100, pageToken = res2.nextPageToken)
+      )
+      // Requesting last page that is empty
+      res4 <- ledger.userManagement.listUsers(
+        ListUsersRequest(pageSize = 100, pageToken = res3.nextPageToken)
+      )
+      // Using not base64 encoded string as a page token
+      onBadTokenError <- ledger.userManagement
+        .listUsers(
+          ListUsersRequest(pageSize = 100, pageToken = UUID.randomUUID().toString)
+        )
+        .mustFail("using not base64 encoded string")
+      // Using negative pageSize
+      onNegativePageSizeError <- ledger.userManagement
+        .listUsers(
+          ListUsersRequest(pageSize = -100, pageToken = "")
+        )
+        .mustFail("using negative page size")
+      // 0 pageSize
+      responseZeroPageSize <- ledger.userManagement.listUsers(
+        ListUsersRequest(pageSize = 0, pageToken = "")
+      )
+    } yield {
+      assert(res1.nextPageToken.nonEmpty, s"First next page token should be non-empty")
+      assertLength("first page", 2, res1.users)
+
+      assert(res2.nextPageToken.nonEmpty, s"Second next page token should be non-empty")
+      assertLength("second page", 3, res2.users)
+
+      assert(res3.nextPageToken.nonEmpty, s"Third next page token should be non-empty")
+      assert(res2.users.nonEmpty, s"Third page should be non-empty")
+
+      assertEquals(
+        s"Last next page token should be empty but was: ${res4.nextPageToken}",
+        res4.nextPageToken,
+        "",
+      )
+      assert(res4.users.isEmpty, s"Last page should be empty but was: ${res4.users}")
+      assertGrpcError(
+        participant = ledger,
+        t = onBadTokenError,
+        expectedCode = Status.Code.INVALID_ARGUMENT,
+        selfServiceErrorCode = LedgerApiErrors.RequestValidation.InvalidArgument,
+        exceptionMessageSubstring = None,
+      )
+      assertGrpcError(
+        participant = ledger,
+        t = onNegativePageSizeError,
+        expectedCode = Status.Code.INVALID_ARGUMENT,
+        selfServiceErrorCode = LedgerApiErrors.RequestValidation.InvalidArgument,
+        exceptionMessageSubstring = None,
+      )
+      assert(
+        responseZeroPageSize.nextPageToken.nonEmpty,
+        "Non-empty page token when pageSize is 0 (and there are some users)",
+      )
+    }
+  })
+
+  test(
+    "TestMaxUsersPageSize",
+    "Exercise max users page size behavior of ListUsers rpc",
+    allocate(NoParties),
+    enabled = _.userManagement.maxUsersPageSize > 0,
+    disabledReason = "requires user management feature with users page size limit",
+  )(implicit ec => { case Participants(Participant(ledger)) =>
+    val maxUsersPageSize = ledger.features.userManagement.maxUsersPageSize
```

Comment:
It's over 4 seconds..

```
- [UserManagementServiceIT:TestMaxUsersPageSize] Exercise max users page size behavior of ListUsers rpc ... Success (4568 ms)
```

Do you think we should make this test case optional and create a new Bazel conformance test target with a lower default page size?


fyi @meiersi-da

Reply:
Our goal is to not have any optional tests in the ledger API test tool.

4 seconds is not that bad. Are we planning on keeping the 10k default max page size? If the plan is to reduce this number in the future, then I would leave it as is.

Reply:
Why don't we lower the max page size to 1k, and thus get a 10x speed improvement at negligible cost?

Reply:
4s is actually pretty bad, given that this test needs to run in isolation, it will add 4s to every conformance test bazel target that we run. I agree with Simon to lower the max page size to 1k

Reply:
Ok, lowering default max page size to 1k